scm/driver/harness/webhook.go

```diff
@@ -356,15 +358,27 @@ func convertPullReq(pr pullReq, ref ref, commit hookCommit) scm.PullRequest {
 	return scm.PullRequest{
 		Number: pr.Number,
 		Title:  pr.Title,
+		Body:   pr.Description,
 		Closed: pr.State != "open",
 		Source: pr.SourceBranch,
 		Target: pr.TargetBranch,
 		Merged: pr.State == "merged",
 		Fork:   "fork",
 		Link:   pr.PrURL,
+		Draft:  pr.IsDraft,
 		Sha:    commit.Sha,
 		Ref:    ref.Name,
 		Author: convertUser(pr.Author),
+		Head: scm.Reference{
+			Name: pr.SourceBranch,
+			Path: scm.ExpandRef(ref.Name, "refs/heads"),
+			Sha:  commit.Sha,
+		},
+		Base: scm.Reference{
+			Name: pr.TargetBranch,
+			Path: scm.ExpandRef(pr.TargetBranch, "refs/heads"),
+			Sha:  pr.MergeBaseSHA,
```

Comment:
mergebasesha isn't necessarily the sha of the targetbranch

Reply:
can you share a screenshot of what you are pointing at, not sure.

Reply:
![Screenshot 2025-04-28 at 1 28 55 PM](https://github.com/user-attachments/assets/b7905b63-6013-4ce5-9915-380174ed69e0)

Reply:
https://www.perplexity.ai/search/what-does-body-and-sha-in-base-UeAsZMv2Q5S9qrkUnKQOjg

Reply:
@abhinav-harness what you are sharing is also pointing to it using `base-sha` for the value, which isn't what your PR is using ( `merge-base-sha` is merge base of head and base, which isn't necessarily the base sha)

Reply:
If you pass `src.TargetRef` then you can use the SHA of that object in the mapping

Reply:
In our code base we have `MergeTargetSHA` field on PR which contains the sha that base ref had when merge check was performed - we could be sending and using that? 🤔

Reply:
i dont think we emit it in webhooks - do we?